src/sqs.rs

```diff
@@ -54,6 +56,82 @@ pub fn get_client(shared_config: &aws_config::Config) -> Result<Client> {
     Ok(Client::from_conf(builder.build()))
 }
 
+const BATCH_SIZE: usize = 10;
+
+/// Send message to a queue from a stream with concurrent invocations of `SendMessageBatch`.
+///
+/// This function retrieves items from the stream until it has exhausted. Any errors
+/// in the stream, or while processing the stream, are returned as soon as they are encountered.
+///
+/// If `concurrency` is `None` then message batches are sent sequentially. A `concurrency` value of
+/// zero, `Some(0)`, is not allowed.
+///
+/// # Example
+///
+/// ```no_run
+/// use aws_config;
+/// use futures::stream;
+/// use cobalt_aws::sqs::{get_client, send_messages_concurrently};
+///
+/// # tokio_test::block_on(async {
+/// let shared_config = aws_config::load_from_env().await;
+/// let client = get_client(&shared_config).unwrap();
+///
+/// let messages = stream::iter(vec![Ok("Hello"), Ok("world")]);
+/// let queue_name = "MyQueue";
+///
+/// // Send up to 4 concurrent API requests at once.
+/// send_messages_concurrently(&client, queue_name, Some(4), messages).await.unwrap();
+/// # })
+/// ```
+///
+/// # Implementation details
+/// This function uses the [SendMessageBatch](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessageBatch.html)
+/// API to send message in batches of 10, which is the maximum allowed batch size.
+pub async fn send_messages_concurrently<Msg: serde::Serialize, St: Stream<Item = Result<Msg>>>(
+    client: &Client,
+    queue_name: &str,
+    concurrency: Option<usize>,
+    msg_stream: St,
+) -> Result<()> {
+    if concurrency == Some(0) {
+        anyhow::bail!("Zero concurrency not allowed.");
+    }
+    let queue_url = client
+        .get_queue_url()
+        .queue_name(queue_name)
+        .send()
+        .await?
+        .queue_url
+        .ok_or(anyhow::anyhow!("Missing queue name"))?;
```

Comment:
```suggestion
        .ok_or_else(|| anyhow::anyhow!("Failed to get queue URL for {queue_name}"))?;
```

The error message here doesn't seem quite right to me, the user did provide a queue name but it somehow doesn't have a corresponding URL.